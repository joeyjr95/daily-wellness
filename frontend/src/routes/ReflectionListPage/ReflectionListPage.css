.ReflectionsListPage{
    padding-top: 12vh;
    display: flex;
    flex-direction: column;
    align-items: center;
}
#archive{
    display: flex;
    flex-direction: column;
    align-items: center;
}

.ReflectionListItem{
  margin-left: auto;
  margin-right: auto;
  list-style: none;
  border: 1px solid white;
  font-size: 1.1em;
  margin-bottom: 3vh;
  padding-left: 50px;
  padding-right: 50px;
  padding-bottom: 10px;
  text-align: center;
  width: 300px;
  border-radius: 4px;
  background-color: #4a50ff;
  word-wrap: break-word;
}
.buttons{
    display: flex;
    flex-direction: column;
}