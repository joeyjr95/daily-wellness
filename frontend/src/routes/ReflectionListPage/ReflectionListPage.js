import React, { Component } from 'react'
import WellnessContext from '../../contexts/WellnessContext'
import WellnessApiService from '../../services/wellness-api-service'
import { Section } from '../../components/Utils/Utils'
import ReflectionListItem from '../../components/ReflectionListItem/ReflectionListItem'
import './ReflectionListPage.css'

export default class ReflectionListPage extends Component {
    static contextType = WellnessContext
    handleClickEdit = () => {
        this.props.history.push(`/edit`)
      }

    componentDidMount() {
        WellnessApiService.getReflections()
            .then(this.context.setReflections)
            .catch(this.context.setError)
    }

    renderReflections() {
        const { reflections = []} = this.context
        return( 
            <>
            <h2 id="archive">Archive</h2>
            <div className="ReflectionList">
                {reflections.map( reflection =>
                <div key={reflection.id} className='ReflectionListItem'>
            <ReflectionListItem
                
                id={reflection.id}
                date_created={reflection.date_created}
                mental_rating={reflection.mental_rating}
                mental_content={reflection.mental_content}
                physical_rating={reflection.physical_rating}
                physical_content={reflection.physical_content}
                handleClickEdit={this.handleClickEdit}
            />
            </div>
        )}
        </div>
        </>
        )
    }
    render(){
        const {error} = this.context
        return(
            <Section list className="ReflectionsListPage">
                {error
                ? <p className='error'> There was an error loading the Reflections, try again</p>
                : this.renderReflections()}
            </Section>
        )
    }
}