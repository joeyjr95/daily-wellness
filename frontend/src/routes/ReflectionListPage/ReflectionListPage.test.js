import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReflectionListPage from './ReflectionListPage'

describe(`ReflectionListPage component`, () => {
  it('renders a .ReflectionListPage by default', () => {
    const wrapper = shallow(<ReflectionListPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Reflection for each notes in array', () => {
    
    const context = {
      reflections: [
        {
            "id": "28",
            "user_id": "1",
            "date_created": "2018-08-15T23:00:00.000Z",
            "mental_rating": "3",
            "mental_content": "stuff",
            "physical_rating": "1",
            "physical_content": "stuffffffff"
        },
        {
            "id": "29",
            "user_id": "1",
            "date_created": "2019-08-15T23:00:00.000Z",
            "mental_rating": "2",
            "mental_content": "stuff",
            "physical_rating": "4",
            "physical_content": "stuffffffff"
        },
      ]
    }
    const div = shallow(<ReflectionListPage />, context)
      .find('div')
    expect(toJson(div)).toMatchSnapshot()
  })
})