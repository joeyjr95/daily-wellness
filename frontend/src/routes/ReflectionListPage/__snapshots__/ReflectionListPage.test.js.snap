// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReflectionListPage component renders a .ReflectionListPage by default 1`] = `
<Section
  className="ReflectionsListPage"
  list={true}
>
  <div>
    <h2>
      Archive
    </h2>
    <ul />
  </div>
</Section>
`;

exports[`ReflectionListPage component renders a Reflection for each notes in array 1`] = `<ul />`;
